blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
                  single_move.move(context)
 
 
-class MoveEngine(object):
-    """MoveEngine provides perturbation functions for the context during the NCMC
-    simulation.
-    Ex.
-        from blues.ncmc import MoveEngine
-        probabilities = [0.25, 0.75]
-        #Where move is a list of two Move objects
-        mover = MoveEngine(move, probabilities)
-        #Get the dictionary of proposed moves
-        mover.moves
-    """
-    def __init__(self, moves, probabilities=None):
-        """Initialize the MovePropsal object that contains functions to perturb
-        the context in the NCMC simulation.
-        Parameters
-        ----------
-        moves : blues.ncmc.Model object or list of n blues.ncmc.Model-like objects
-            Specifies the possible moves to be performed.
+class SmartDartMove(RandomLigandRotationMove):
+    def __init__(self, structure, basis_particles, dart_size=0.2*unit.nanometers, resname='LIG'):
+        super(SmartDartMove, self).__init__(structure, resname=resname)
+        self.dartboard = []
+        self.n_dartboard = []
+        self.particle_pairs = []
+        self.particle_weights = []
+        self.basis_particles = basis_particles
+        self.dart_size = dart_size
+        self.calculateProperties()
+
+    def dartsFromParmEd(self, coord_files, topology=None):
+        """
+        Used to setup darts from a generic coordinate file, through MDtraj using the basis_particles to define
+        new basis vectors, which allows dart centers to remain consistant through a simulation.
+        This adds to the self.n_dartboard, which defines the centers used for smart darting.
+        Arguments
+        ---------
+        system: simtk.openmm.system
+            Openmm System corresponding to the system to smart dart.
+        coord_files: list of str
+            List containing coordinate files of the system for smart darting.
+        basis_particles: list of 3 ints
+            Specifies the 3 indices of particles whose coordinates will be used
+            as basis vectors. If None is specified, uses those found in basis particles.
+            If None uses self.basis_particles instead.
+        Returns
+        -------
+        n_dartboard: list of 1x3 np.arrays
+            Center of mass coordinates of atom_indices particles in new basis set.
```

Comment:
One last thing on the doc string: What's the difference between providing a topology and not? Should be documented (and right now the doc string doesn't even say this is a possible argument).